### ISSUE TITLE
Panic when opening old indexes

### ISSUE THREAD

**Issue by gyk** [quickwit-inc/tantivy#1754]
**Describe the bug**

Tantivy panics when reading an index of old version with no `DOC_STORE_VERSION` field in the footer:

```
thread 'MainLoop' panicked at 'actual doc store version: 2222222222, expected: 1', tantivy-0.19.0\src\store\footer.rs:30:13
```

**Comment by PSeitz** [comment]
@gyk There was a breaking change in the format for `0.19`. We don't support reading older versions

**Comment by gyk** [comment]
It would be better to return an `Err` rather than panicking.

**Comment by cjrh** [comment]
@PSeitz is there a guarantee on compatibility for patch version changes? e.g., will all versions in 0.19.X have a guarantee to work on the same index structure?

**Comment by PSeitz** [comment]
Yes, patch versions will work on the same index. They only contain bugfixes
